plasmapy/dispersion/numerical/kinetic_alfven_.py

```diff
@@ -0,0 +1,303 @@
+    T_i : `~astropy.units.Quantity`
+        The ion temperature in units of :math:`K` or :math:`eV`.
+    theta : `~astropy.units.Quantity`, single valued or 1-D array
+        The angle of propagation of the wave with respect to the
+        magnetic field, :math:`\cos^{-1}(k_z / k)`, in units must be
+        convertible to :math:`rad`. Either single valued or 1-D array
+        of size :math:`M`.
+    gamma_e : `float` or `int`, optional
+        The adiabatic index for electrons, which defaults to 1.  This
+        value assumes that the electrons are able to equalize their
+        temperature rapidly enough that the electrons are effectively
+        isothermal.
+    gamma_i : `float` or `int`, optional
+        The adiabatic index for ions, which defaults to 3. This value
+        assumes that ion motion has only one degree of freedom, namely
+        along magnetic field lines.
+    z_mean : `float` or int, optional
+        The average ionization state (arithmetic mean) of the ``ion``
+        composing the plasma.  Will override any charge state defined
+        by argument ``ion``.
+
+    Returns
+    -------
+    omega : Dict[str, `~astropy.units.Quantity`]
+        A dictionary of computed wave frequencies in units
+        :math:`rad/s`.  The dictionary contains a key for each:
+        theta value provided. The value for each key will be
+        a :math:`N × M` array.
+
+    Raises
+    ------
+    TypeError
+        If applicable arguments are not instances of
+        `~astropy.units.Quantity` or cannot be converted into one.
+
+    TypeError
+        If ``ion`` is not of type or convertible to
+        `~plasmapy.particles.particle_class.Particle`.
+
+    TypeError
+        If ``gamma_e``, ``gamma_i``, or``z_mean`` are not of type
+        `int` or `float`.
+
+    ~astropy.units.UnitTypeError
+        If applicable arguments do not have units convertible to the
+        expected units.
+
+    ValueError
+        If any of ``B``, ``k``, ``n_i``, ``T_e``, or ``T_i`` is negative.
+
+    ValueError
+        If ``k`` is negative or zero.
+
+    ValueError
+        If ``ion`` is not of category ion or element.
+
+    ValueError
+        If ``B``, ``n_i``, ``T_e``, or ``T_I`` are not single valued
+        `astropy.units.Quantity` (i.e. an array).
+
+    ValueError
+        If ``k`` or ``theta`` are not single valued or a 1-D array.
+
+    Notes
+    -----
+    Using the 2 × 2 matrix approach method from :cite:t:`bellan:2012`,
+    this function computes the corresponding wave frequencies in units
+    :math:`rad/s`. This approach comes from :cite:t:`hasegawa:1982`,
+    :cite:t:`morales:1997` and :cite:t:`william:1996`; who argued that
+    a 3 × 3 matrix that describes warm plasma waves, is able to be
+    represented as a 2 × 2 matrix because the compressional
+    (i.e., fast) mode can be factored out. This results in the
+    determinant, when in the limit of
```

Comment:
```suggestion
    (i.e., fast) mode can be factored out. The result is that the
    determinant, when in the limit of
```
Could you double check that the suggestion is still right?

Reply:
It does look good to me! 👍🏻